src/storage-controller/src/lib.rs

```diff
@@ -3049,7 +2870,13 @@ where
     // TODO(guswynn): we need to be more careful about the update time we get here:
     // <https://github.com/MaterializeInc/materialize/issues/25349>
```

Comment:
As an aside: can we guarantee at this point, that the old storage controller is fenced out, and that the upper from `.collections_frontier()` below is linearizably the greatest upper?

Reply:
I don't know! I think maybe we never could, and I think certainly information about uppers can always be outdated.

Reply:
I think ideally we need to ensure another writer to a shard is fenced...ill have to come back to this